src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebProxy.cs

```diff
@@ -18,6 +18,24 @@ internal WebProxy(Uri address)
             _proxyAddress = address;
         }
 
+        public override bool Equals(object obj) => Equals(obj as WebProxy);
+
+        public override int GetHashCode() => HashCode.Combine(_proxyAddress, _credentials, BypassProxyOnLocal);
+
+        public bool Equals(WebProxy other)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+
+            // _proxyAddress cannot be null as it is set in the constructor            
+            return GetHashCode() == other.GetHashCode()
+                && other._credentials == _credentials
+                && _proxyAddress.Equals(other._proxyAddress)
+                && BypassProxyOnLocal == other.BypassProxyOnLocal;
```

Comment:
Do we really need calculate and compare hashes?

Reply:
I did not have that in there initially. I added in response to someone's suggestions. I can remove it if you like.

Reply:
Remove it 👍

Reply:
Done